ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -244,6 +245,7 @@ public void configureBaseApplication(
         new JacksonMessageBodyProvider(JsonMapper.INSTANCE.mapper);
     config.register(jsonProvider);
     config.register(JsonParseExceptionMapper.class);
+    config.register(new KsqlRestServiceContextBinder(ksqlConfig));
```

Comment:
How will this work when we need to configure it differently for each request?

Reply:
The binder will bind the factory using the class type instead of the instance of the class. Like:
```
bindFactory(KsqlRestServiceContextFactory.class).to(ServiceContext.class);
```

This way, the binder will make sure to create a new `KsqlRestServiceContextFactory` for every new REST request. I then can get the user token and other security credentials using dependency injection in the factory.

One thing, though, is that I cannot add the `KsqlConfig` with this code. The only way I will do it is by calling a static method of the factory to set this during the Binder initialization. After that, every new instance of the Factory will have the ksqlConfig set + the user credentials.